frontend/src/component/ui/Modal.spec.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { Modal } from './Modal';
+
+jest.useFakeTimers();
+
+it('renders without crashing', () => {
+  render(<Modal />);
+});
+
+it('opens and closes correctly', () => {
```

Comment:
Let's separate open and close into two test cases so that when one of the failed, we know exactly what failed:
opens correctly

Reply:
closes modal when click on close button

Reply:
done